definitions/features/candlepin_database.rb

```diff
@@ -0,0 +1,60 @@
+class Features::CandlepinDatabase < ForemanMaintain::Feature
+  CANDLEPIN_DB_CONFIG = '/etc/candlepin/candlepin.conf'.freeze
+
+  include ForemanMaintain::Concerns::BaseDatabase
+
+  metadata do
+    label :candlepin_database
+
+    confine do
+      file_exists?(CANDLEPIN_DB_CONFIG)
+    end
+  end
+
+  def configuration
+    @configuration || load_configuration
+  end
+
+  def cpdb_help_output
+    @cpdb_help_output || run_cpdb_help_cmd
+  end
+
+  def check_option_using_cpdb_help(option_name, parent_cmd = '')
+    parent_cmd = '/usr/share/candlepin/cpdb' if parent_cmd.empty?
+    help_cmd = "#{parent_cmd} --help |  grep -c '\\-\\-#{option_name} '"
+    execute(help_cmd).strip.to_i
+  end
+
+  def execute_cpdb_validate_cmd
+    main_cmd = cpdb_validate_cmd
+    unless main_cmd.empty?
+      main_cmd += " -u #{configuration['username']} -p '#{configuration[%(password)]}'"
```

Comment:
I would recommend using `shellescape` helper here (that we have defined in the `SystemHelpers`, and also wrapping the `username` argument into `''`. Perhaps additional helper in the system helpers, that would take the args like this and turn them into proper command would be useful for other cases as well:

```ruby
main_cmd += format_shell_args('-u' => configuration['username'], '-p' => configuration['password'])
```